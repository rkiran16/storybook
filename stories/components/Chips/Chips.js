import { html, nothing } from "lit-html";


export const Chip = ({checked, disabled, icon= 'done', label = "Choice Chip", inputType = 'checkbox', showLeftIcon = false, showRightIcon = false}) => {
   return html`
      <div class="chip">
         <input type=${inputType} class="btn-check" id=${`chip-${icon}`} ?disabled=${disabled} ?checked=${checked} autocomplete="off">
         <label class="btn rounded-pill" for=${`chip-${icon}`}>
             <span class=${`material-icons me-2 ${!showLeftIcon || checked? 'icon' : ''}`}>${icon}</span>
             ${label}
            ${showRightIcon ? html`<span class="material-icons ms-2">close</span>` : nothing} 
         </label>
      </div>`
}

export const FilterChip = () => {
   return html`
      <div class="chip chip--filter">
         <input type="checkbox" class="btn-check" id='filter-chip' autocomplete="off">
         <label class="btn rounded-pill" for='filter-chip'>
            Filter
            <span class="material-icons icon"></span>
         </label>
      </div>`
}