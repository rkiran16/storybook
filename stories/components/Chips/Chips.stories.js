import { Chip, FilterChip } from "./Chips";
import { html } from 'lit-html';

export default {
   title: "Components/Chips",
   parameters: {
      layout: 'centered',
      zeplinLink: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/62e09078d2dcad13d00cf021",
    },
   argTypes: {
      checked: {
         name: "Selected",
         type: { name: 'boolean', required: false},
         description: "Chip selected state",
         defaultValue: false,
         table: {
            type: { summary: 'boolean' },
            defaultValue: { summary: false },
         },
         control: {type: 'boolean'}
      },
      showLeftIcon: {
         name: "Leading Icon",
         type: { name: 'boolean', required: false},
         description: "Show Leading Icon",
         defaultValue: false,
         table: {
            type: { summary: 'boolean' },
            defaultValue: { summary: false },
         },
         control: {type: 'boolean'}
      },
      showRightIcon: {
         name: "Trailing Icon",
         type: { name: 'boolean', required: false},
         description: "Show Trailing Icon",
         defaultValue: false,
         table: {
            type: { summary: 'boolean' },
            defaultValue: { summary: false },
         },
         control: {type: 'boolean'}
      },
      disabled: {
         name: "Disabled",
         type: { name: 'boolean', required: false},
         description: "Use when the user cannot proceed or until an input is collected",
         defaultValue: false,
         table: {
           type: { summary: 'boolean' },
           defaultValue: { summary: false },
         },
         control: {
           type: 'boolean'
         }
      },
      inputType: {
         name: "Input Type",
         defaultValue: 'checkbox',
         control: { type: 'radio' },
         description: "Select Input Type",
         table: {
            type: { summary: 'radio' },
            defaultValue: { summary: 'checkbox' },
         },
         options: ['checkbox', 'radio'],
      },
      icon: {
         name: "Icon",
         control: { type: 'select'},
         description: "Select Icon",
         defaultValue: "done",
         options: ['default','sort', 'place', 'bike'], 
         table: {
            type: { summary: 'select' },
            defaultValue: { summary: 'done' },
         },
         mapping: {
            default: 'done',
            sort: 'filter_alt',
            place: 'place',
            bike: 'directions_bike'
         },
      }
   }
}

const StandardTemplate = args => Chip(args);
// Standard Chip
export const Standard = StandardTemplate.bind({});
// Selected Chip
export const SelectedChip = StandardTemplate.bind({});
SelectedChip.args = {
    checked: true
}
// Sort & Filter Chip
export const SortAndFilterChip = StandardTemplate.bind({});
SortAndFilterChip.args = {
    showLeftIcon: true,
    icon: "filter_alt",
    label: "Sort & Filter"
}

// Filter Chip
const FilterChipTemplate = args => FilterChip(args);
export const FilterChips = FilterChipTemplate.bind({});
FilterChips.args = {}


//Chips Row
const ChipsRowTemplate = () => {
   return html`
      <div class="d-flex">
        <div class="me-2">${Chip({label: "Choice Chip 1", icon: 'alarm', showLeftIcon: true})}</div>
        <div class="me-2">${Chip({label: "Choice Chip 2", icon: 'place', showLeftIcon: true})}</div>
        <div class="me-2">${Chip({label: "Choice Chip 3", icon: 'search', showLeftIcon: true})}</div>
        <div class="me-2">${Chip({label: "Choice Chip 4", icon: 'directions_bike', showLeftIcon: true})}</div>
      </div>
    `
}

export const ChipsRow = ChipsRowTemplate.bind({});
ChipsRow.args = {}