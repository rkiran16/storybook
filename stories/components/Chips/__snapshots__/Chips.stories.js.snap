// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Chips Chips Row 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
      <div class=\\"d-flex\\">
        <div class=\\"me-2\\">",
    "</div>
        <div class=\\"me-2\\">",
    "</div>
        <div class=\\"me-2\\">",
    "</div>
        <div class=\\"me-2\\">",
    "</div>
      </div>
    ",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div class=\\"chip\\">
         <input type=",
        " class=\\"btn-check\\" id=",
        " ?disabled=",
        " ?checked=",
        " autocomplete=\\"off\\">
         <label class=\\"btn rounded-pill\\" for=",
        ">
             <span class=",
        ">",
        "</span>
             ",
        "
            ",
        " 
         </label>
      </div>",
      ],
      "values": Array [
        "checkbox",
        "chip-alarm",
        undefined,
        undefined,
        "chip-alarm",
        "material-icons me-2 ",
        "alarm",
        "Choice Chip 1",
        Symbol(lit-nothing),
      ],
    },
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div class=\\"chip\\">
         <input type=",
        " class=\\"btn-check\\" id=",
        " ?disabled=",
        " ?checked=",
        " autocomplete=\\"off\\">
         <label class=\\"btn rounded-pill\\" for=",
        ">
             <span class=",
        ">",
        "</span>
             ",
        "
            ",
        " 
         </label>
      </div>",
      ],
      "values": Array [
        "checkbox",
        "chip-place",
        undefined,
        undefined,
        "chip-place",
        "material-icons me-2 ",
        "place",
        "Choice Chip 2",
        Symbol(lit-nothing),
      ],
    },
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div class=\\"chip\\">
         <input type=",
        " class=\\"btn-check\\" id=",
        " ?disabled=",
        " ?checked=",
        " autocomplete=\\"off\\">
         <label class=\\"btn rounded-pill\\" for=",
        ">
             <span class=",
        ">",
        "</span>
             ",
        "
            ",
        " 
         </label>
      </div>",
      ],
      "values": Array [
        "checkbox",
        "chip-search",
        undefined,
        undefined,
        "chip-search",
        "material-icons me-2 ",
        "search",
        "Choice Chip 3",
        Symbol(lit-nothing),
      ],
    },
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div class=\\"chip\\">
         <input type=",
        " class=\\"btn-check\\" id=",
        " ?disabled=",
        " ?checked=",
        " autocomplete=\\"off\\">
         <label class=\\"btn rounded-pill\\" for=",
        ">
             <span class=",
        ">",
        "</span>
             ",
        "
            ",
        " 
         </label>
      </div>",
      ],
      "values": Array [
        "checkbox",
        "chip-directions_bike",
        undefined,
        undefined,
        "chip-directions_bike",
        "material-icons me-2 ",
        "directions_bike",
        "Choice Chip 4",
        Symbol(lit-nothing),
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Chips Filter Chips 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
      <div class=\\"chip chip--filter\\">
         <input type=\\"checkbox\\" class=\\"btn-check\\" id='filter-chip' autocomplete=\\"off\\">
         <label class=\\"btn rounded-pill\\" for='filter-chip'>
            Filter
            <span class=\\"material-icons icon\\"></span>
         </label>
      </div>",
  ],
  "values": Array [],
}
`;

exports[`Storyshots Components/Chips Selected Chip 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
      <div class=\\"chip\\">
         <input type=",
    " class=\\"btn-check\\" id=",
    " ?disabled=",
    " ?checked=",
    " autocomplete=\\"off\\">
         <label class=\\"btn rounded-pill\\" for=",
    ">
             <span class=",
    ">",
    "</span>
             ",
    "
            ",
    " 
         </label>
      </div>",
  ],
  "values": Array [
    "checkbox",
    "chip-done",
    false,
    true,
    "chip-done",
    "material-icons me-2 icon",
    "done",
    "Choice Chip",
    Symbol(lit-nothing),
  ],
}
`;

exports[`Storyshots Components/Chips Sort And Filter Chip 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
      <div class=\\"chip\\">
         <input type=",
    " class=\\"btn-check\\" id=",
    " ?disabled=",
    " ?checked=",
    " autocomplete=\\"off\\">
         <label class=\\"btn rounded-pill\\" for=",
    ">
             <span class=",
    ">",
    "</span>
             ",
    "
            ",
    " 
         </label>
      </div>",
  ],
  "values": Array [
    "checkbox",
    "chip-filter_alt",
    false,
    false,
    "chip-filter_alt",
    "material-icons me-2 ",
    "filter_alt",
    "Sort & Filter",
    Symbol(lit-nothing),
  ],
}
`;

exports[`Storyshots Components/Chips Standard 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
      <div class=\\"chip\\">
         <input type=",
    " class=\\"btn-check\\" id=",
    " ?disabled=",
    " ?checked=",
    " autocomplete=\\"off\\">
         <label class=\\"btn rounded-pill\\" for=",
    ">
             <span class=",
    ">",
    "</span>
             ",
    "
            ",
    " 
         </label>
      </div>",
  ],
  "values": Array [
    "checkbox",
    "chip-done",
    false,
    false,
    "chip-done",
    "material-icons me-2 icon",
    "done",
    "Choice Chip",
    Symbol(lit-nothing),
  ],
}
`;
